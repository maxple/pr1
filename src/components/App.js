import { Route, Switch } from 'react-router-dom'
import Menu from './ui/Menu'
import Whoops404 from './ui/Whoops404'
import { ColorContainer, ColorsContainer, NewColorContainer } from './containers'
import '../stylesheets/APP.scss'

const App = () =>
  <Switch>
    <Route exact
           path="/:id"
           component={ColorContainer} />
    <Route path="/"
           component={({ match, location }) => (
             <div className="app">
               <Menu sort={location.pathname.replace('/sort/', '')} />
               <NewColorContainer />
               <Switch>
                 <Route exact
                        path="/"
                        component={ColorsContainer} />
                 <Route path="/sort/:sort"
                        component={ColorsContainer} />
                 <Route component={Whoops404} />
               </Switch>
             </div>
           )} />
  </Switch>

export default App
