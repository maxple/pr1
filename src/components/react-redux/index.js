import React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import App from './components/App'
import storeFactory from './store'

const store = storeFactory()

window.React = React

const root_render = () => render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root'),
)

export default root_render