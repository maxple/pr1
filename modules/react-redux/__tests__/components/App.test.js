import renderer from 'react-test-renderer'
import App from '../../components/App'
import React from 'react'
import {Provider} from 'react-redux'
import storeFactory from '../../store'

const store = storeFactory()

window.React = React

describe('App', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
