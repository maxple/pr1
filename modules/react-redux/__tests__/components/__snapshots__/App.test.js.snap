// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app"
>
  <nav
    className="menu"
  >
    <h1>
      Sort Colors
    </h1>
    <a
      className="selected"
      href="#"
      onClick={[Function]}
    >
      date
    </a>
    <a
      className={null}
      href="#"
      onClick={[Function]}
    >
      title
    </a>
    <a
      className={null}
      href="#"
      onClick={[Function]}
    >
      rating
    </a>
  </nav>
  <form
    className="add-color"
    onSubmit={[Function]}
  >
    <input
      placeholder="color title..."
      required={true}
      type="text"
    />
    <input
      required={true}
      type="color"
    />
    <button>
      ADD
    </button>
  </form>
  <div
    className="color-list"
  >
    <section
      className="color"
    >
      <h1>
        Ocean Blue
      </h1>
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
          />
        </svg>
      </button>
      <div
        className="color"
        style={
          Object {
            "backgroundColor": "#0070ff",
          }
        }
      />
      <div
        className="time-ago"
      >
        3/13/2016
      </div>
      <div>
        <div
          className="star-rating"
        >
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <p>
            3
             of 
            5
             stars
          </p>
        </div>
      </div>
    </section>
    <section
      className="color"
    >
      <h1>
        Tomato
      </h1>
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
          />
        </svg>
      </button>
      <div
        className="color"
        style={
          Object {
            "backgroundColor": "#d10012",
          }
        }
      />
      <div
        className="time-ago"
      >
        3/11/2016
      </div>
      <div>
        <div
          className="star-rating"
        >
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <p>
            2
             of 
            5
             stars
          </p>
        </div>
      </div>
    </section>
    <section
      className="color"
    >
      <h1>
        Lawn
      </h1>
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
          />
        </svg>
      </button>
      <div
        className="color"
        style={
          Object {
            "backgroundColor": "#67bf4f",
          }
        }
      />
      <div
        className="time-ago"
      >
        3/10/2016
      </div>
      <div>
        <div
          className="star-rating"
        >
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <div
            className="star"
            onClick={[Function]}
          />
          <p>
            1
             of 
            5
             stars
          </p>
        </div>
      </div>
    </section>
    <section
      className="color"
    >
      <h1>
        Party Pink
      </h1>
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
          />
        </svg>
      </button>
      <div
        className="color"
        style={
          Object {
            "backgroundColor": "#ff00f7",
          }
        }
      />
      <div
        className="time-ago"
      >
        3/9/2016
      </div>
      <div>
        <div
          className="star-rating"
        >
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <div
            className="star selected"
            onClick={[Function]}
          />
          <p>
            5
             of 
            5
             stars
          </p>
        </div>
      </div>
    </section>
  </div>
</div>
`;
